core-kotlin-modules/core-kotlin-files/src/main/kotlin/com/baeldung/tempfile/TempFile.kt

```diff
@@ -0,0 +1,52 @@
+package com.baeldung.tempfile
+
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.createTempFile
+import kotlin.io.path.writeText
+
+fun createTempFileWithKotlinExtensions(): Path {
+    val tempFile = createTempFile(prefix = "kotlinTemp", suffix = ".tmp")
+    println("Temporary file created with Kotlin extensions at: ${tempFile.toAbsolutePath()}")
+
+    tempFile.writeText("Kotlin Path Data")
+
+    return tempFile
```

Comment:
Let's also get the delete on exit line into the example, even if we don't show it all at once.

```suggestion
    tempFile.writeText("Kotlin Path Data")

    tempFile.toFile().deleteOnExit()

    return tempFile
```